### PR TITLE
Bugfix (3.4): Text::readable_list

### PR DESCRIPTION
**Pull request by slik**
Fixed the data for `test_readable_list` and `test_readable_list` itself.
Also replaced `InvalidArgumentException` with `Kohana_Exception` because `Kohana_Exception` is the enter point of all Kohana exceptions. Tell me if I'm wrong.

### CODE REVIEW

**Comment by lenton** [comment]
I've tested this PR and it fixes the existing four failures caused by TextTest, +1 for merge!

File: classes/Kohana/Text.php [hunk 1]
```diff
@@ -705,12 +705,12 @@ public static function readable_list(array $words, $conjunction = 'and', $serial
 			// Check that the word isn't an array itself.
 			if (is_array($word))
 			{
-				throw new InvalidArgumentException('The array must only have one dimension.');
+				throw new Kohana_Exception('The array must only have one dimension.');
```
Comment (shadowhand) [inline classes/Kohana/Text.php:708]: why the change in exception type?

Comment (slik) [inline classes/Kohana/Text.php:708]: @shadowhand 
The first reason is that `Kohana_Exception` is expected in the test, the second reason is in my comment to the PR: because Kohana_Exception is the enter point of all Kohana exceptions.
